include/swift/AST/DiagnosticsSIL.def

```diff
@@ -173,6 +173,9 @@ ERROR(immutable_value_passed_inout,none,
 ERROR(assignment_to_immutable_value,none,
       "immutable value '%0' may not be assigned to",
       (StringRef))
+ERROR(mutating_protocol_witness_method_on_let_constant,none,
+      "cannot perform mutating instruction: '%0' is a 'let' constant",
```

Comment:
Those both definitely sound better. SIL is an implementation detail, so it's better for diagnostics to talk in terms of user-facing concepts.